web-frontend/src/hooks/useUserDetails.ts

```diff
@@ -1,18 +1,23 @@
 import {useContext, useEffect, useState} from "react";
 import {UserDetails} from "../model/UserDetails";
+import {Subscription} from "../model/Subscription";
 import {AuthContext} from "../context/AuthProvider";
 import {toast} from "react-toastify";
-import {getUserDetails} from "../service/api-service";
+import {getSubscriptions, getUserDetails} from "../service/api-service";
 
 export default function useUserDetails(){
     const [userDetails, setUserDetails] = useState<UserDetails>()
+    const [subscriptions, setSubscriptions] = useState<Subscription[]>()
     const {token} = useContext(AuthContext)
 
     useEffect(() =>{
         getUserDetails(token)
             .then(details => setUserDetails(details))
-            .catch(() => toast.error("Connection failed. Please retry."))
+            .catch(() => toast.error("Connection failed to get user details. Please retry."))
+        getSubscriptions(token)
+            .then(subs => setSubscriptions(subs))
```

Comment:
Why is there an extra hook for subscriptions? Can't you just update your model in frontend and get the subscribtions with the userDetails?

Reply:
I like the hook: In future I can give the hook to separate subscripiton components.